src/serialize.rs

```diff
@@ -0,0 +1,103 @@
+use super::*;
+use serde::{Serialize, de::DeserializeOwned};
+
+#[cfg(test)]
+mod test{
+    const ACTION_SELECT: &str = "SELECT";
+    const ACTION_SUPER_SELECT: &str = "SUPER_SELECT";
+    const ACTION_AWESOME_SUPER_SELECT: &str = "AWESOME_SUPER_SELECT";
+    const TESTPATH : &str = "test.keymap";
```

Comment:
Might be useful to add these files to .gitignore so they aren't accidentally committed.